cpu/native/socket_zep/socket_zep.c

```diff
@@ -350,26 +350,70 @@ void socket_zep_setup(socket_zep_t *dev, const socket_zep_params_t *params)
         }
     }
     real_freeaddrinfo(ai);
+
     if (res < 0) {
         err(EXIT_FAILURE, "ZEP: Unable to bind socket");
     }
-    dev->sock_fd = res;
-    ai = NULL;
+
+    return res;
+}
+
+static int _connect_remote(socket_zep_t *dev, const socket_zep_params_t *params)
+{
+    int res;
+    static const struct addrinfo hints = { .ai_family = AF_UNSPEC,
+                                           .ai_socktype = SOCK_DGRAM };
+    struct addrinfo *ai = NULL, *remote;
+
+    if (params->remote_addr == NULL) {
+        return -1;
+    }
+
     if ((res = real_getaddrinfo(params->remote_addr, params->remote_port, &hints,
                                 &ai)) < 0) {
         errx(EXIT_FAILURE, "ZEP: unable to get remote address: %s\n",
              gai_strerror(res));
     }
+
     for (remote = ai; remote != NULL; remote = remote->ai_next) {
         if (real_connect(dev->sock_fd, remote->ai_addr, remote->ai_addrlen) == 0) {
             break;  /* successfully connected */
         }
     }
+
     if (remote == NULL) {
         err(EXIT_FAILURE, "ZEP: Unable to connect socket");
     }
+
     real_freeaddrinfo(ai);
 
+    return res;
+}
+
+void socket_zep_setup(socket_zep_t *dev, const socket_zep_params_t *params)
+{
+    int res;
+
+    DEBUG("socket_zep_setup(%p, %p)\n", (void *)dev, (void *)params);
+    assert((params->remote_addr != NULL) && (params->remote_port != NULL));
+
+    memset(dev, 0, sizeof(socket_zep_t));
+    dev->netdev.netdev.driver = &socket_zep_driver;
+
+    res = _bind_local(params);
+
+    if (res < 0) {
+        dev->sock_fd = socket(AF_INET6, SOCK_DGRAM, 0);
+    } else {
+        dev->sock_fd = res;
+    }
+
+    _connect_remote(dev, params);
+
+    /* send dummy data to connect to dispatcher */
+    /* (no valid ZEP header, will by dropped in _recv() */
+    real_write(dev->sock_fd, "HELLO", 5);
```

Comment:
Even though this is not "standard compliant" to ZEP (I am not aware of a standard though and my implementation is based on the [wireshark dissector code](https://github.com/boundary/wireshark/blob/master/epan/dissectors/packet-zep.c#L25-L43) and what little the [wireshark doc provides](https://www.wireshark.org/docs/dfref/z/zep.html)) I would prefer if this packet is recognizable as a ZEP packet. For ZEPv2 there are [two types known: ACK and DATA](https://github.com/boundary/wireshark/blob/master/epan/dissectors/packet-zep.c). However, the ZEP type field is one byte, so enough to space to invent other types.

How about defining (driver internally, not in the global `sys/include/net/zep.h` header) a new type

```C
#define SOCKET_ZEP_V2_TYPE_HELLO   (255)
```

The packet could be the same format as DATA (as the wireshark dissector [always falls back to DATA when the packet is not an ACK](https://github.com/boundary/wireshark/blob/07eade8124fd1d5386161591b52e177ee6ea849f/epan/dissectors/packet-zep.c#L150-L158)) but leave all fields 0, except for the `reserved` field which could contain the "HELLO" string.

This way the packet is disected in a wireshark dump as a ZEP packet, and also clearly visable as a hello packet.

Reply:
Ok, dropped it from this PR.

My reasoning was that the packet should be easy to distinguish as non-ZEP packet and dropped early.

But I can also do the version with the ZEP header if you think that's cleaner.

Reply:
IMHO it makes the purpose of this packet clearer if just looking at a dump when it is a ZEP packet.